src/app/approve/approve.component.ts

```diff
@@ -247,6 +248,17 @@ export class ApproveComponent implements OnInit {
         const groupKeyName = this.transaction.metadata.messagingGroupKeyName
         description = `register group key with name "${groupKeyName}"`
         break
+      case TransactionMetadataDAOCoinLimitOrder:
+        let buyingDAOCoinCreatorPublicKey = this.base58KeyCheck(this.transaction.metadata.buyingDAOCoinCreatorPublicKey);
+        if (buyingDAOCoinCreatorPublicKey == "") {
+          buyingDAOCoinCreatorPublicKey = "DESO";
+        }
+        let sellingDAOCoinCreatorPublicKey = this.base58KeyCheck(this.transaction.metadata.sellingDAOCoinCreatorPublicKey);
+        if (sellingDAOCoinCreatorPublicKey == "") {
+          sellingDAOCoinCreatorPublicKey = "DESO"
+        }
+        description = `create DAO coin limit order to sell ${sellingDAOCoinCreatorPublicKey} ` +
+          `to buy ${buyingDAOCoinCreatorPublicKey}`
```

Comment:
```suggestion
        description = `create DAO coin limit order to sell ${sellingDAOCoinCreatorPublicKey} ` +
          `to buy ${buyingDAOCoinCreatorPublicKey}`
```

We want this description to say `sell LazyNina DAO Coin` if it's not selling DESO. Can you make that adjustment? Additionally, it would be great if we could include the price and quantity. I know that's tricky, but we want to make it as clear as possible what the user is authorizing.

Reply:
Also, we need to update the description if this is cancelling an order.

Reply:
you can see how we parse hexNanoStrings to get quantity from line 241. Figuring out scaled exchange rate will be a bit trickier, but I believe in you :)